internal/clock/clock.go

```diff
@@ -49,7 +49,7 @@ func (m *Mock) Add(d time.Duration) {
 	// Calculate the final time.
 	end := m.now.Add(d)
 
-	for len(m.timers) > 0 && m.now.Before(end) {
+	for len(m.timers) > 0 && (m.now.Before(end) || m.now.Equal(end)) {
```

Comment:
this is the bug I mentioned above.

Reply:
can we fix this but in its own diff? ideally it would contain only this change and an additional test that makes sure the boundary is accounted for

Reply:
#41 drops our internal fork of the clock altogether.

I could raise a separate PR to fix the bug ... but I don't think adding tests (which currently don't exist at all :D) would make sense.

Reply:
Okay, after a slack chat I merged #41 into this PR.